docs/core/data-model/DataModel.md

```diff
@@ -0,0 +1,144 @@
+
+```
+GenericDto dto = new GenericDto();
+dto.provider = "provider1";
+dto.service = "service";
+dto.service = "resource";
+dto.value = 42;
+
+push.pushUpdate(dto);
+```
+
+In this approach the sensiNact core will automatically create a model for `provider1` if it doesn't already exist. The model will automatically be extended with a service called `service` and a resource called `resource` if needed, and then the resource will be set with a value of 42.
+
+In fact the process becomes more obvious if the `model` field of the `GenericDto` is set. This indicates the model that should be used for the provider and allows different providers to share the same model.
+
+> [!IMPORTANT]
+> All automatically created resources are set to be `UPDATABLE`, meaning that they can be updated by pushing data updates, but not using a `SET` operation.
+
+#### Modifying the model
+
+When using a data first approach the model can always be updated and dynamically expanded. This means that the sending a new update dto targeting a different resource in an existing provider will implicitly extend the model for that provider to include any new service or resource as needed.
+
+This process gives maximum flexibility, but it is important to remember that any mistakes when pushing updates will potentially lead to unexpected content in the model for your provider.
+
+### The code first approach
+
+The code first approach is similar to the data first approach in that no formal model is created, and that users simply push data updates into sensiNact using the `PrototypePush` service. The key difference is that unlike the data first approach users define one or more custom DTOs representing the resources that they want to update. These DTOs form the template for the model, with the field names and type information being used to generate the model as updates are received.
+
+For example, the following DTO:
+
+```
+@Model("WeatherStation")
+public class WeatherStation {
+  @Provider
+  public String provider;
+
+  @Service("power")
+  @Data
+  public Integer status;
+
+  @Service("weather")
+  @Data
+  public Double temperature;
+
+  @Service("weather")
+  @Data
+  public Double humidity;
+
+}
+```
+
+This DTO defines a model called `WeatherStation`. It supplies the `Provider` id in the `provider` field, and defines two services `power` and `weather`. The `power` service has a `status` resource of type `Integer`, and the `weather` service has resources called `temperature` and `humidity` both of type `Double`.
+
+In this approach the sensiNact core will automatically create a model for `provider1` if it doesn't already exist. The model will automatically be extended with a service called `service` and a resource called `resource` if needed, and then the resource will be set with a value of 42.
+
```

Comment:
I think that we should remove this line, since the line before perfectly explains the DTO.